src/lib/db/project-store.ts

```diff
@@ -265,25 +266,21 @@ class ProjectStore implements IProjectStore {
             await this.db(TABLE)
                 .where({ id: data.id })
                 .update(this.fieldToRow(data));
-            if (
-                data.defaultStickiness !== undefined ||
-                data.featureLimit !== undefined
-            ) {
-                if (await this.hasProjectSettings(data.id)) {
-                    await this.db(SETTINGS_TABLE)
-                        .where({ project: data.id })
-                        .update({
-                            default_stickiness: data.defaultStickiness,
-                            feature_limit: data.featureLimit,
-                        });
-                } else {
-                    await this.db(SETTINGS_TABLE).insert({
-                        project: data.id,
-                        default_stickiness: data.defaultStickiness,
-                        feature_limit: data.featureLimit,
-                        project_mode: 'open',
+
+            if (await this.hasProjectSettings(data.id)) {
+                await this.db(SETTINGS_TABLE)
+                    .where({ project: data.id })
+                    .update({
+                        default_stickiness: data.defaultStickiness || null,
+                        feature_limit: data.featureLimit || null,
```

Comment:
We might risk overriding a value here... if one of these is not in the payload we should leave it undefined and not touch it

Reply:
Addressed in https://github.com/Unleash/unleash/pull/5317/commits/6803bce9925689988579f39ab8c635a25d835ac7 - Let me know what you think of it.